tests/test_waittask.py

```diff
@@ -0,0 +1,101 @@
+from asyncio import ensure_future
+import asyncio
+from time import time
```

Comment:
this should be `perf_counter`, time isn't guaranteed to not go backwards. I suggest a helper function that gives you ms resolution so that you don't need to do the conversion — all the wait stuff is specified in ms
```suggestion
from time import perf_counter

def perf_counter_ms():
    return perf_counter()*1000
```